### DISCUSSION TITLE
[Mania] General lazer Scoring feedback survey, Questions, Results and Conclusions

### DISCUSSION THREAD

**Discussion by Adrriii** [ppy/osu#26314] [Ideas]
For the past two weeks, we have been preparing, relaying and summarizing a general survey with the following goals:
- Bring interest to osu!lazer in the mania community
- Collect opinions regarding debated scoring topics
- Facilitate and structure focused feedback

Using an instance of the [form-auth](https://git.cartooncraft.fr/osufrlive/form-auth) micro-service, we've been able to group opinions by skill ranges in an attempt to differentiate what a more casual player base might want from osu!lazer as opposed to a more competitive one.

Shared in the development forums and various communities, the survey had great reception, nearing 400 verified submissions from 60  countries in a 3 days span.

**In a short summary, the results of this survey should bring comfort to the dev team for having taken decisions that mostly match the expectations of all skill groups.**

The entirety of answers, stats and annotations can be found on the [spreadsheet](https://docs.google.com/spreadsheets/d/10ymvvtIVhHbkOAnCcnDHQ4RGFSZyhkLjKcIyU0tfRFE).

---

# Survey metrics

(see the Poll Stats tab)

- Verified Submissions: 387
- Countries: 60
- Average and median osu!mania rank : 25k / 3.5k

Skill group shares:
- 5 digits and higher : 24.50%
- 4 digits : 52%
- 3 digits and lower : 20.50%
- Unranked at the time : 3%

On each question, if a skill group appears to be an outlier, a mention will be made. However please note that each of these skill groups share the same majority, most if not all the time.

Answers should then be from high and diverse enough sample to conclude on general opinions of each skill group with generally low error margin.

# Survey

## Score standardization

This section contains two questions, both aimed at collecting the sentiment towards:
1- The 1 million cap itself
2- The 0 to 1 million scoring range

The reasons for asking such questions being the possibility of breaking these in lazer in favor of new scoring mechanics.

![image](https://github.com/ppy/osu/assets/16978299/1a9ac8a3-7dd0-4e64-8dc9-3ecefa188858)

Ranging from 1 to 5 in the scale of the preference of keeping this cap, here were the global and per-skill range results:
![image](https://github.com/ppy/osu/assets/16978299/aafa2d86-0a70-43dc-a82d-4049d7be06a3)

The sentiment towards keeping the 1M cap is very pronounced, and it is the main subject of the free textfield remarks. (see column 'AK' in the 'FormV2' sheet)

This sentiment is especially shared among higher ranked players, but we can still very well observe it among the more casual playerbase.

**In conclusion: The statement that the 1M cap is 'sacred' among the community can be verified, or at the very least that it's vastly preferred. Many would be disappointed if not kept.**

![image](https://github.com/ppy/osu/assets/16978299/ea6a8fe8-8b5b-405c-afa0-c2fcbef23880)

Ranging from 1 to 5 in how useful the 0 to 1 million score range is useful in judging a score, here were the global and per-skill range results:
![image](https://github.com/ppy/osu/assets/16978299/9b5f7e2d-3396-464b-9bdb-f5bf4d022a9e)

These results show a rather strong reliance on the 0-1M range to judge a score, as all-across 40% of players claim to be unable to judge a score, were the limit not relevant anymore.

Another third of players would be inconvenienced, bringing the preference of keeping such a range to a very high 75%.

**In conclusion: To avoid throwing players off and keep consistency with scoring, it is a much safer solution to avoid scoring mechanics breaking the 0-1M range.**

## SS criteria

This section is exclusively aimed at collecting opinions and ideas about SS criteria. The unique question asks the player what would suit him in general and allows him to express an alternative. The player can select as many options as he likes.

![image](https://github.com/ppy/osu/assets/16978299/820fe76d-4332-4253-9e3f-abe1d6bc0a57)

Results:
![image](https://github.com/ppy/osu/assets/16978299/efcf2f46-f702-4e9d-9603-c4764b73d6e2)

Unsurprisingly, a vast majority of players are still fine with the current stable SS criteria for score v1.

However, we find very surprising results in that more than a third are actually ok with keeping the score v2 mechanic, consistently across all skill levels. This still tells us that two thirds are not ok with it.

A very extreme result seeing that only 0.5% do not have an opinion, choosing that any system would suit them.

Only a minority are open to the Accuracy based judgement, admittedly vaguely defined (some Other propositions go in more details, citing 99.5% for example).

As already discussed as not happening, many propositions rely on an additional rank (SSS, SS+, X, ...)

**In conclusion:
There is a strong preference towards the scorev1 SS criteria of a Perfect FC (all Perfect/Great). However, if a reliable solution cannot be offered to match this expectation, a compromise of the currently implemented Score v2 SS (1,000,000) would still fit a portion of the playerbase.**

All manual propositions can be found in the column 'N' of the 'FormV2' sheet.

## Lazer leaderboards : Default and Tabs

This next section holds a quite complex question, aimed at understanding the sentiment towards various possible new leaderboards either as Default, or as a separate Tab and whether score should be multiplied. Examples of other rhythm games leaderboards are shown to present PP / multiplied score and accuracy leaderboards.

![image](https://github.com/ppy/osu/assets/16978299/1f1cbca5-3e74-453e-81cb-145448fd772f)

Results:
![image](https://github.com/ppy/osu/assets/16978299/3547431d-9192-42e1-97d0-f526be24ed33)
![image](https://github.com/ppy/osu/assets/16978299/340513dc-5a2b-4218-ae79-a10d216aae02)

These results show that the mania community is very open to having more options for leaderboards. While there is a very strong preference for keeping the 1million score tab the default one, players tend to have their own preference, confirming the idea that allowing players to have their own default tab (eg. remembering the last tab across game sessions) would be beneficial.

We can however note a rather marked reticence towards the existence of options such as Multiplied score and Experimental leaderboards, with still two thirds of the player base either agreeing with them as options or simply not having an opinion.

**In conclusion: The base 1 million score tab should remain the standard and default leaderboard as per strong player preference, and the existence of multiple tabs that anyone can chose to use is received positively.**
*If too many tabs is to be avoided, the following tabs seem to be more popular, in order : Selected mods, PP, Accuracy, Experimental, Multiplied score.*

## Mods Ranking

Lazer seeing new mods and mods customization, many seemed excited to discuss rankability of mania mods. So we explored community opinions regarding preference only.

![image](https://github.com/ppy/osu/assets/16978299/bca5d5b2-ea51-4fc0-8a95-8b3412cec37f)

Results:
![image](https://github.com/ppy/osu/assets/16978299/0bf4429c-d45a-4ec7-b74d-1e55df64ecb5)
![image](https://github.com/ppy/osu/assets/16978299/3e7267fb-1344-4557-a0d2-445e9efdc432)

These results show a strong welcome for ranking custom rates, with the exception of the 1.5\~2.0 range showing mixed results and especially the 0.5\~0.75 range that is still majorly agreed upon but "only" by two thirds of players.

Hitwindow mods show very mixed results, overall not being clearly liked as Ranked, however we can note a pronounced agreement towards HR from the more casual playerbase, EZ staying majorily unwanted.

As for beatmap-modifying mods, the results show an unsurprising majority of negative sentiment towards their Ranked status, while we can note a small push for the Inverted LN mod.

**In conclusion: Custom rates seem overall welcomed in the mania community. HR and EZ don't seem to be safely rankable with only such a poll, and beatmap-modifying mods are majorily unwanted.**

## "Lazer Experiments"

Since the current state of lazer doesn't show what has been tried, we attempted to explain the concepts of "Perfect bonus score" and the possibility of a DT/NC multiplier as to gather sentiment towards these, if they should be tried in a tab or in the default leaderboard.

The results are not very telling since most people chose to ignore that answer or remained neutral. Still, here is the visualization of the results just in case.

![image](https://github.com/ppy/osu/assets/16978299/5fc0a4d9-e115-4a5a-a825-8fce566f4bfe)
![image](https://github.com/ppy/osu/assets/16978299/c39ee8d4-cbfb-4b76-88f4-c369535a78b5)

These results for this optional question are admittedly not very telling, especially since a sizeable portion of respondants didn't pick a side.
The goal was to observe how welcome "lazer experiments" are among the default leaderboard versus inside a tab.
It seems like most players would rather leave the default leaderboard untouched but would only mostly agree for these experiments being shown in a tab. Ignoring neutral opinions, the clear outline is the DT multiplier which players would prefer seeing in a separate leaderboard.
However, as seen in previous results, it is not a particularily popular one and PP could serve this purpose.

**In conclusion: It may be preferable to keep experiments in a related tab, if it is deemed useful to start one.**

# Conclusions

## Score standardization

It seems like a vast majority of players would be happy with the decision of keeping the mania scoring as it was. There should be no need to revisit this core concept, and more focus can be aimed at the existing combo/acc parameters without including bonus. Furthermore, it is a strong indicator that concepts such as bonus could indeed have a bad reception if attempted live, even if balanced.

## SS criteria

This problematic is seemingly the most important one that hasn't fully been resolved in lazer, however these results show that it may not be as important as previously thought. At least, solving it may cause further problems than keeping the current score v2 solution.

## Leaderboards

The current vision of lazer seems to correlate with the results : Especially if the user's leaderboard tab is kept upon restarting the game. While the default score tab makes consensus, many players seem to have their personal favorite leaderboard.

## Mods Ranking

Players seem to mostly agree on having adjustable rates in the ranking section, with small doubts about more extreme rates. Timing window mods seem justifiable while the rest do not.

## Lazer Experiments

The mania community seems to be partly open to having an experiment tab as proposed by smoogipoo. It does not make consensus, but most either agree or are neutral.

# Discussion

Thanks for taking the time to read this. We hope it will be useful, at least in comforting the dev team that mania looks promising from most player's end.
Any feedback and remarks are appreciated, and we remain available. A discord server has been set up around osu!mania in lazer to hopefully collect more feedback and help people transitioning.

**Comment by cdwcgt** [comment]
Another very important issue here is the score value of perfect and great.
according to [ppy said on the server](https://discord.com/channels/188630481301012481/1187357282208587777/1189160804289871993)

> make PERFECT/MAX300s give bonus score again (but only 1 point extra to make sure they don't outweigh normal plays)

In some tournaments, the scores are generally close to SS. At this time, the proportion of `perfect` will be very important.

if we make perfect only award 1 extra point, that means if any 1x200 will break the advantage with high `perfect` ratio.
mean 1x100 another all perfect cannot beat all great score (unless object num very large)

This may also be an undesirable outcome

**Reply by Adrriii** [reply]
FYI this isn't planned anymore following instant mania players feedback after the poll 👍

**Comment by fishy15** [comment]
Do you think that making bonus score a fixed proportion of the overall score would still work? [I mentioned this earlier](https://github.com/ppy/osu/discussions/25810#discussioncomment-7916757), but since it was said earlier that SS requirements would match SV1, I think it was deemed unnecessary. But since it seems like SS needs to be 100% acc, then this idea still might be feasible, though I don't know how much this differs from SV2. 

Also not sure if infrastructure supports having bonus be a fixed portion of score

**Reply by PatouZZ** [reply]
I don't think SS must be a 100% accuracy, for some Vertical Scrolling Rhythm Game, a rank matches with what hits you get in your play.
For example, in this case for scorev1 in stable, a SS is a 100% BUT it has 2 different judgements in it which are 300MAX and 300.
The 300 hit isn't a perfect one and you still have the best rank which seems kinda strange on some point.
That's why our proposal is to keep this idea with all 300MAX/300 hits to have a SS rank and why not add one for the full 300MAX play, which correspond to a 1M score (for example X or SSS)

**Reply by Adrriii** [reply]
The additional grade idea was rejected multiple times and won't happen afaik.

The bonus being a fixed proportion might be slightly better balance wise but maybe it starts complicating things while still going against opinion. If it were say between 990k and 1M maybe, but then any SS would beat an S regardless of ratio which seems very unbalanced.

**Reply by fishy15** [reply]
>The 300 hit isn't a perfect one and you still have the best rank which seems kinda strange on some point.

My personal opinion is that it is better to have 300 and MAX contribute different to accuracy since it is closer to reality, but I think this is the detail that is easiest to compromise on since it matches SV1. So if we must have SS be based on accuracy, then it is necessary that MAX contributes to bonus score.

>If it were say between 990k and 1M maybe, but then any SS would beat an S regardless of ratio which seems very unbalanced.

As calculated above, i think making it worth 50k points is roughly equivalent to the current scoring system so it should act roughly the same? However, it is not exactly the same as SV2 since bonus score scales linearly instead of impacting accuracy, which doesn't grow linearly, and I'm not entirely sure what problems this could cause. If there is interest then I could look how much it differs from the current scoring system, but I'm not entirely sure how to implement this/use the tooling to see how this impact scores. 

This being said, I do have some concerns about the apparent complexity of the scoring system, but I will let others have the final say on this matter.

**Reply by Adrriii** [reply]
It seems to me like the bonus system only applies for raw values though. Your solution would work with such a larger portion but indeed it seems like a complex task to standardize something that isn't meant to be. It seemed to me like the idea of differentiating 'accuracy score' judgement weights from actual 'accuracy' weighs would be much easier.

**Comment by RushFTK** [comment]
i guess we should also consider development & standard with other modes, as far as i know i haven't find reason why we must keep 1000k as a cap while other modes could break 1000k caps. so i also vote dosen't matter in that. (only finds goods on development.)
but for how many bonus point here, i **strongly** disagreed with 1 score per note since it's compelete broken accuracy gameplay style. so i sugeest using `1.8% * 1000K(standard score cap) * 85% (accuracy% in standard score cap)` as total bonus score which could keep things similar with current score system.

**Reply by Adrriii** [reply]
> i haven't find reason why we must keep 1000k as a cap while other modes could break 1000k caps

I think it's more : What is a good enough reason to break this cap for mania, when players strongly expect it to stay ? Other game modes have clear bonus mechanics (Spinners, Drum rolls, Bananas), but not mania.

If the bonus system is implemented and standardized to remain below 1M, then the problem seems solved, however it may just be a more complex solution than just using different weights for acc than acc score. (see the question just above suggesting a similar idea)
